test/fail_compilation/test16195.d

```diff
@@ -0,0 +1,14 @@
+/*
+ * TEST_OUTPUT:
+---
+fail_compilation/test16195.d(13): Error: delete p is unsafe
+fail_compilation/test16195.d(13): Error: delete p is impure
+fail_compilation/test16195.d(13): Error: delete p is non-@nogc
```

Comment:
Having a different error message here is quite confusing.

I suggest something along the lines of:
```
fail_compilation/test16195.d(13): Error: pure function 'test' cannot use impure 'delete' operator
fail_compilation/test16195.d(13): Error: safe function 'test' cannot use system 'delete' operator
fail_compilation/test16195.d(13): Error: @nogc function 'test' cannot use non-@nogc 'delete' operator
```